### DISCUSSION TITLE
Is there any way to left join with non unique join frame?

### DISCUSSION THREAD

**Discussion by soufisouf** [h2oai/datatable#3495] [Q&A] [answered]
Pleaaase help :(

I have two DF, DF1 (col A, B, C)  and DF2, (col, A, B, D)

I want to left join DF1 with DF2 in order to have ALL THE ROWS of DF1, and in front of each, the equivalent in DF2.

The issue is that DF2 contains duplicated by A and B and I need to duplicate them in the resulting DF.

Can anyone help please?
thanks a lot

**Comment by samukweku** [comment]
hi @soufisouf at the moment this is not possible - the left dataframe should have unique keys

**Reply by soufisouf** [reply]
@samukweku thanks a lot. So the unique solution is to convert to pandas, do the join, and convert back to datatable? :(

**Reply by samukweku** [reply, marked as answer]
Yes, that is the alternative. Or convert to arrow and use pyarrow/polars

**Reply by soufisouf** [reply]
Thanks a lot
